File: src/main/java/work/lclpnet/ap2/game/panda_finder/PandaFinderMiniGame.java
package work.lclpnet.ap2.game.panda_finder;

import net.minecraft.item.ItemStack;
import net.minecraft.item.Items;
import net.minecraft.util.Formatting;
import net.minecraft.util.Identifier;
import work.lclpnet.ap2.api.game.*;
import work.lclpnet.ap2.base.ApConstants;
import work.lclpnet.ap2.base.ArcadeParty;
import work.lclpnet.kibu.translate.text.FormatWrapper;

public class PandaFinderMiniGame implements MiniGame {

    @Override
    public Identifier getId() {
        return ArcadeParty.identifier("panda_finder");
    }

    @Override
    public GameType getType() {
        return GameType.FFA;
    }

    @Override
    public String getAuthor() {
        return ApConstants.PERSON_LCLP;
    }

    @Override
    public ItemStack getIcon() {
        return new ItemStack(Items.BAMBOO);
    }

    @Override
    public boolean canBeFinale(GameStartContext context) {
        return true;
    }

    @Override
    public boolean canBePlayed(GameStartContext context) {
        return true;
    }

    @Override
    public MiniGameInstance createInstance(MiniGameHandle gameHandle) {
        return new PandaFinderInstance(gameHandle);
    }

    @Override
    public Object[] getDescriptionArguments() {
        return new Object[] { FormatWrapper.styled(PandaFinderInstance.WIN_SCORE, Formatting.YELLOW) };
    }
}

File: src/minigames/panda_finder/src/main/kotlin/work/lclpnet/ap2/game/panda_finder/PandaFinderMiniGame.kt
package work.lclpnet.ap2.game.panda_finder

import net.minecraft.ChatFormatting
import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.game.GameStartContext
import work.lclpnet.ap2.api.game.GameType
import work.lclpnet.ap2.game.MapLevelGameFactory
import work.lclpnet.ap2.game.MiniGame
import work.lclpnet.kibu.translate.text.FormatWrapper

class PandaFinderMiniGame : MiniGame {
    override val id = ApConstants.identifier("panda_finder")
    override val type = GameType.FFA
    override val author = ApConstants.PERSON_LCLP
    override fun getIcon(manager: RegistryAccess) = ItemStack(Items.BAMBOO)
    override fun canBeFinale(context: GameStartContext) = true
    override fun canBePlayed(context: GameStartContext) = true
    override fun createFactory() = MapLevelGameFactory(::PandaFinderInstance)
    override val descriptionArguments: Array<Any> = arrayOf(
        FormatWrapper.styled(WIN_SCORE, ChatFormatting.YELLOW)
    )
}
